src/js/reducers/deviceReducer.js

```diff
@@ -179,9 +179,10 @@ const deviceReducer = (state = initialState, action) => {
       };
     }
     case DeviceConstants.SET_DEVICE_FILTERS: {
+      const filters = action.filters.filter(filter => filter.key && filter.operator && filter.scope && typeof filter.value !== 'undefined');
```

Comment:
small sidenote: the `typeof ... !== 'undefined'` check most likely is not necessary as `&&` should treat `undefined` as a falsey value - `... && filter.value`

Reply:
@mzedel yup, but zero is false, isn't it? the "doesn't" exist is "$exists: 0"

Reply:
ah right, when the filters are selected by the user, the value be 1 and only mapped to 0 before being sent to the backend, but when the filters are retrieved from the backend the value will be 0, but defined - sorry I missed that! 👌